### ISSUE TITLE
Let's make Dislikes visible (Crowd sourced rating system)

### ISSUE THREAD

**Issue by WeslyG** [ajayyy/SponsorBlock#1039]
https://www.theverge.com/2021/11/10/22773299/youtube-dislike-button-hide-public-count-numbers-small-creator-protection

Today I found out that YouTube is going to disable the public display of dislikes. I think this is an unfair and destructive decision. 
It seems to me that I am not the only one judging by many videos (most often educational) by the number of dislikes on them. 
The sponsorblock community is huge, and it seems to be the best place today to add a dislike button right here.

Yes, I understand that this is not the direct functionality of sponsorblock, but it seems unjustified to make another application for these purposes.

What do you think about it?

**Comment by ajayyy** [comment]
Still not really sure about it as it might attract abuse to the anonymous userID system

**Comment by ajayyy** [comment]
I think of this were to be a serious feature, it would need to be more powerful than like/dislike. We would need to define some categories of content to label it as. I feel that would actually be useful and be more inline with the extension

**Comment by ajayyy** [comment]
It could be combined with full video sponsor/self promo reporting as another category

**Comment by usermyname12** [comment]
Maybe we can add an estimation. ((likes+views/100)/views*1500)  i've tested it and it works *ok* on videos with average views. It overvalues the good ones but it definitely catches the bad ones (unless it is very popular or small). If we can tweak the 100 above so more views=smaller that number then it can be useful to big videos as well.

**Comment by ShadowTheAge** [comment]
This is not the functionality of this app, but the app for likes/dislikes must be popular to work, so this one seems appropriate.
Maybe show in-app likes/dislikes for everyone but require authentication to post a like/dislike. (A non-binary "like/dislike" will be better but not a priority)

**Comment by ajayyy** [comment]
I've thought about this more, and think this should be split off to a seperate but connected project. It can use the same backend ([SponsorBlockServer](https://github.com/ajayyy/SponsorBlockServer)), but have different extension front-ends.

SponsorBlock:
_Fixing_ the experience of watching videos on YouTube (things with clear answers)

- Removing as much advertisement from videos as possible
- Skipping to the part of the video you want to see
- Labeling entire videos as ads/self promo (that still should be in SponsorBlock)
- (Coming very soon) crowdsourced chapters 
- (Potential Future) relabeling titles and thumbnails to more useful information
- (Potential Future) tl;dr and other one-answer crowd-sourced things

New Extension:
_Finding and labeling_ the things you actually want to see (the vote-based multiple-answer stuff)

- A custom like/dislike system
- A system to report videos as containing specific categories of subjecting information (incorrect info, misleading, etc.)
- (Potential future) filtering for the videos you actually will care about based off crowdsourced data

I think SponsorBlock should remain only with things that have one answer, and not things where a voting % is more useful. This new project is much more risky, and I don't want it to drag down SponsorBlock.

There are other advantages to launching it as a new extension such as being able to work on all sites instead of just YouTube, which doesn't work with the permissions SponsorBlock has.

It is still possible get the advantage of SponsorBlock's userbase by telling people about it if they click the dislike button.

Thoughts?

**Comment by CarrotIsACarrot** [comment]
it good 👍‍

**Comment by leodoi3** [comment]
Someone should create an independent dislike database + browser extension that will work on the same principle as SponsorBlock. It might be even possible to fetch and archive the current dislike counts, since API should remain working till December 13th.

**Comment by ajayyy** [comment]
@leodoi3 that's fine as well, but I just want to be clear, I'm working on one myself now

**Comment by OpeningCredits** [comment]
> I've thought about this more, and think this should be split off to a separate but connected project. It can use the same backend ([SponsorBlockServer](https://github.com/ajayyy/SponsorBlockServer)), but have different extension front-ends.
> 
> SponsorBlock: _Fixing_ the experience of watching videos on YouTube (things with clear answers)
> 
>     * Removing as much advertisement from videos as possible
> 
>     * Skipping to the part of the video you want to see
> 
>     * Labeling entire videos as ads/self promo (that still should be in SponsorBlock)
> 
>     * (Coming very soon) crowdsourced chapters
> 
>     * (Potential Future) relabeling titles and thumbnails to more useful information
> 
>     * (Potential Future) tl;dr and other one-answer crowd-sourced things
> 
> 
> New Extension: _Finding and labeling_ the things you actually want to see (the vote-based multiple-answer stuff)
> 
>     * A custom like/dislike system
> 
>     * A system to report videos as containing specific categories of subjecting information (incorrect info, misleading, etc.)
> 
>     * (Potential future) filtering for the videos you actually will care about based off crowdsourced data
> 
> 
> I think SponsorBlock should remain only with things that have one answer, and not things where a voting % is more useful. This new project is much more risky, and I don't want it to drag down SponsorBlock.
> 
> There are other advantages to launching it as a new extension such as being able to work on all sites instead of just YouTube, which doesn't work with the permissions SponsorBlock has.
> 
> It is still possible get the advantage of SponsorBlock's userbase by telling people about it if they click the dislike button.
> 
> Thoughts?
> 
> (More discussion also on #dislikes on [Discord](https://discord.gg/sponsorblock) and [Matrix](https://matrix.to/#/!JQKEeNpQLJqeEDobVx:ajay.app?via=ajay.app))

This seems to make the most sense. A seperate extention that somehow still makes use of the SponsorBlock platform and it's popularity without directly affecting it.

I also really like the idea of being able to "label" a video, this would be exceptionally useful for labeling videos that are clickbait, fake news, toxic. I don't know how the ui would look, but perhaps you could click a drop down menu with a selection of buttons, each one representing what the content of the video might contain, then have the most clicked buttons appear under the video, or something like that.

Ultimately I'm not a software designer so you will know the best way to implement these features, but I do think these features are very useful and an extention like this could be very popular, especially if it's somehow able to piggyback off SponsorBlock and it's popularity somehow.

**Comment by BlackSails7** [comment]
In my opinion, If it works with SponsorBlock, I'd say add it there with an option to disable it. If it doesn't work properly, (because users would be able to login with multiple accounts and add tons of fake likes/dislikes) then create a new add-on which requires more permissions to prevent that.

> "A system to report videos as containing specific categories of subjecting information (incorrect info, misleading, etc.)"

 but this is based on each person's (objective) opinion. You can literally label incorrect info/misleading ANYTHING you disagree with. So is it really useful? Most people are not rational. They are religious in so many ways. Anything that threatens their beliefs they automatically label as misinformation or misleading.

Another thing I should add is that we should be able to see the likes/dislikes **bar** from the thumbnails of the videos BEFORE clicking to watch them. There are scripts and add-ons which did this (until now).

https://greasyfork.org/en/scripts/30254-youtube-ratingbars-like-dislike-rating
https://addons.mozilla.org/en-US/firefox/addon/youtube-thumbnail-rating-bar

 I think adding the likes/dislikes right NOW is what needs to be done(because everyone is thinking about it now) The add-on can be updated later to add other features.
 
The like/dislike button-icons must be both dark and white theme friendly. Should be the same as the ones youtube uses probably, so that people are already used to them!

**Comment by Rjevski** [comment]
> Fixing the experience of watching videos on YouTube (things with clear answers)

I'd argue that being able to dislike and see dislikes is a "fix" for the experience.

Whether those dislikes are accurate or not shouldn't really be SponsorBlock's responsibility (the YT ones aren't either; Google is absolutely incompetent at filtering out spam/bots and even then, human click farms can be hired for cheap).

I don't think "diluting" the SponsorBlock userbase is a good idea with two extensions. The extension needs network effects to work so the more users have it installed the better. It's also good for the user experience; I'd argue that seeing dislikes and blocking sponsored trash is both in the interests of the user and if they want one they'd like appreciate the other, so having both in a single & easy to install extension would be my preference.

**Comment by OpeningCredits** [comment]
> > Fixing the experience of watching videos on YouTube (things with clear answers)
> 
> I'd argue that being able to dislike and see dislikes is a "fix" for the experience.
> 
> Whether those dislikes are accurate or not shouldn't really be SponsorBlock's responsibility (the YT ones aren't either; Google is absolutely incompetent at filtering out spam/bots and even then, human click farms can be hired for cheap).
> 
> I don't think "diluting" the SponsorBlock userbase is a good idea with two extensions. The extension needs network effects to work so the more users have it installed the better. It's also good for the user experience; I'd argue that seeing dislikes and blocking sponsored trash is both in the interests of the user and if they want one they'd like appreciate the other, so having both in a single & easy to install extension would be my preference.

I somewhat agree with this. But then it becomes a matter of trying to merge these two features in a way that doesn't detract from the core aspect of sponsorblock and what is primarily used for as a tool.

**Comment by BlackSails7** [comment]
apparently dislikes are visible if you are not logged in?!

**Comment by dubesor** [comment]
> 
> 
> Maybe we can add an estimation. ((likes+views/100)/views*1500) i've tested it and it works _ok_ on videos with average views. It overvalues the good ones but it definitely catches the bad ones (unless it is very popular or small). If we can tweak the 100 above so more views=smaller that number then it can be useful to big videos as well.

spits out completely false and random results. I checked on my videos and 98% liked videos became 14% etc with your formula. Engagement needs to be included for that you only have comments, which can get deleted. So, there is no way  with the accessible data to determine like/dislike ratio.

As for including an own upvote/downvote system I disagree because it's not within scope of the extension, nor the extension name. I can already see an endless set of youtube related features being added over time totally alienating what the extension set out to do initially (feature creep).

**Comment by OpeningCredits** [comment]
> but this is based on each person's (objective) opinion. You can literally label incorrect info/misleading ANYTHING you disagree with. So is it really useful? Most people are not rational. They are religious in so many ways. Anything that threatens their beliefs they automatically label as misinformation or misleading.


One way we can avoid potential bias is by having the most clicked on labels appear higher on the list with greater percentages. For example; Clickbait: 98%, fake news 89% Misleading 80% etc. Then a viewer can decide for themselves if they want to trust those figures or keep watching.

**Comment by leodoi3** [comment]
> > but this is based on each person's (objective) opinion. You can literally label incorrect info/misleading ANYTHING you disagree with. So is it really useful? Most people are not rational. They are religious in so many ways. Anything that threatens their beliefs they automatically label as misinformation or misleading.
> 
> One way we can avoid potential bias is by having the most clicked on labels appear higher on the list with greater percentages. For example; Clickbait: 98%, fake news 89% Misleading 80% etc. Then a viewer can decide for themselves if they want to trust those figures or keep watching.

How would this work if it were political commercials and rallies? The viewer would just watch it anyway and not care for the labels.

**Comment by OpeningCredits** [comment]
> How would this work if it were political commercials and rallies? The viewer would just watch it anyway and not care for the labels.

What's the problem here?

**Comment by ajayyy** [comment]
Non-binding name brainstorm: https://pollunit.com/en/polls/bx80e7zrqfanagkhr2sl5w

**Comment by Anarios** [comment]
> Someone should create an independent dislike database + browser extension that will work on the same principle as SponsorBlock. It might be even possible to fetch and archive the current dislike counts, since API should remain working till December 13th.

That's the idea behind my project - http://returnyoutubedislike.com/

Chrome\Firefox extension, and a DB behind it (currently scraping youtube API to save as much data as possible)

Volunteer creators could log in into extension to disclose their channel's dislike counts.

Integration with SponsorBlock and\or YoutubeVanced, and any other related project would be awesome.

**Comment by BlackSails7** [comment]
> 
> 
> > Someone should create an independent dislike database + browser extension that will work on the same principle as SponsorBlock. It might be even possible to fetch and archive the current dislike counts, since API should remain working till December 13th.
> 
> That's the idea behind my project - http://returnyoutubedislike.com/
> 
> Chrome\Firefox extension, and a DB behind it (currently scraping youtube API to save as much data as possible)
> 
> Volunteer creators could log in into extension to disclose their channel's dislike counts.
> 
> Integration with SponsorBlock and\or YoutubeVanced, and any other related project would be awesome.

Yeah, it's good to have both. Thanks for your work, but could you bring the like/dislike ratio bar back, and also make it visible from the thumbnails? There are some scripts/addons which used to work (as I liked them above before)
https://greasyfork.org/en/scripts/30254-youtube-ratingbars-like-dislike-rating
https://addons.mozilla.org/en-US/firefox/addon/youtube-thumbnail-rating-bar


But who knows for how long your addon/script will work. Especially if it got popular, youtube would find a way to disable it.

We absolutely need an independent system. This way channels will no longer have the power to hide their video ratings.

We also need independent comments tbh. Too many people have begun disabling comments sections. And deleting comments that hurts them.

**Comment by ajayyy** [comment]
An idea brought about in discussion in Discord:

Instead of having upvote and downvote, have positive and negative report options. Then, these options can be displayed in a bar with colours, similar to GitHub languages. There can be a seperator in the middle to seperate the "positive" side from the "negative" side.

![image](https://user-images.githubusercontent.com/12688112/141891945-f180205c-0c5d-4b0c-9caa-0d4196faf414.png)

**Comment by OpeningCredits** [comment]
> An idea brought about in discussion in Discord:
> 
> Instead of having upvote and downvote, have positive and negative report options. Then, these options can be displayed in a bar with colours, similar to GitHub languages. There can be a seperator in the middle to seperate the "positive" side from the "negative" side.
> 
> ![image](https://user-images.githubusercontent.com/12688112/141891945-f180205c-0c5d-4b0c-9caa-0d4196faf414.png)
> 
> I think this will be a lot more flexible, while still giving us the positive/negative ratio for quick glances.
> 
> GitHub UI for reference: ![image](https://user-images.githubusercontent.com/12688112/141892006-1db32ccf-8e5f-4e63-a18d-4a512d0df118.png)

I'm on the fence about this one. Do you mean a sliding scale with multiple options such as "slightly dislike", "dislike", "slightly like", and "like"? As someone who doesn't use Github for pretty much anything other than what I'm doing now, and the fact that most add-on users are probably the same, I'm not really sure how much of a hit this would be aesthetically.

**Comment by BlackSails7** [comment]
> 
> 
> An idea brought about in discussion in Discord:
> 
> Instead of having upvote and downvote, have positive and negative report options. Then, these options can be displayed in a bar with colours, similar to GitHub languages. There can be a seperator in the middle to seperate the "positive" side from the "negative" side.
> 
> ![image](https://user-images.githubusercontent.com/12688112/141891945-f180205c-0c5d-4b0c-9caa-0d4196faf414.png)
> 
> I think this will be a lot more flexible, while still giving us the positive/negative ratio for quick glances.
> 
> GitHub UI for reference: ![image](https://user-images.githubusercontent.com/12688112/141892006-1db32ccf-8e5f-4e63-a18d-4a512d0df118.png)

I think you could add this as an option (in addon settings) for anyone who wants it. But by default it should be the basic like/dislike in my opinion. So I think we can have both. Most people want the simple like/dislike, this is what we are all used to. For the rest of users who want more details, they can enable that bar in settings, and also this would give them more flexibility when they rate videos.

But it depends on what the options in that bar would be. If it's just "partially dislike", "dislike", "partially like", "like" then you can sum all the negative ones into the dislikes, and all the positives into the likes. (for those who don't use the bar)

If the bar has options for labels (scam/misleading/useful/etc)... then can you have both? I guess maybe if you sum all the negative ones into the "dislike" and the positives into the like (for those who don't use the bar)

Its a good idea, my only problem is that it's probably not convenient. Because when you only have like/dislike it's just 1 click. When you have too many options you cant even decide what to choose, because most of the time it's multiple of them.

I simply don't know. It's just something we need to test out first. Like/dislikes we already know how it works.

**Comment by ajayyy** [comment]
No, not slightly dislike. It would be positive and negative reasons (misleading, scam, useful, etc.)

**Comment by OpeningCredits** [comment]
> No, not slightly dislike. It would be positive and negative reasons (misleading, scam, useful, etc.) since like/dislike is can be pretty meaningless

That does make more sense, and I see what you mean now. Like or disliking something based on a reason. That sounds pretty smart. My only suggestion there would be to keep the feature as simple as possible so that it doesn't use much memory.

**Comment by BlackSails7** [comment]
What about a simple like/dislike system, but when you click like or dislike, it would require a reason by popping up a small panel above the buttons.

Then when people hover the bar under these 2 buttons (like/dislike) it would show the labels with different colors and when you hover them it would display the text.

But what if you simply like a video because you agree with it (a video that you wouldn't necessarily call "useful" or "interesting") Or what if the video is simply "funny"? I guess we would need many labels to choose from.

Labels I can think of for the like button : [like, useful, interesting, unreal, exciting, funny]
Labels for the dislike button: [dislike, misleading title, scam, annoying, harmful, propaganda]

So the buttons would be a simple like/dislike, but when you click each, it would popup a horizontal panel above with these options (like a context menu, same as it is on '[messenger reactions](https://tse2.mm.bing.net/th?id=OIP.UFpIep9vQpsJ2DBKTlRz2wHaDJ)' but with text not emojis) and these would appear in a ratio bar bellow the like/dislike buttons like it is on [Github like you said](https://user-images.githubusercontent.com/12688112/141892006-1db32ccf-8e5f-4e63-a18d-4a512d0df118.png). So that's a possible idea.

btw the addon should not be removing youtube's likes/dislikes (or at least have the option to keep them)

**Comment by TrashAccount3424** [comment]
Someone I trying to do this here https://github.com/pgamerx/yt-dislikes-viewer/issues/6
Maybe join forces?

**Comment by pgamerx** [comment]
> Someone is trying to do this here https://github.com/pgamerx/yt-dislikes-viewer/issues/6
> Maybe join forces?

Hi, 
I have been working on the API for this that will allow us to make a `POST` request  that adds certain data to the database. For eg- `?video_id=69&action=dislike/like`. If SponsorBlock actually is ready to integrate with the api, I will change the API as per the above comments. The API will also be able to get the data using `GET` request which will return a JSON object.

As far as storage is concerned, this API is currently on replit and I am using sqlite,but I do have a VPS where I decided to host it once it's fully complete. We can try to store data in the shortest form possible instead of strings.

**Comment by pgamerx** [comment]
The main problem for me was Oauth because I cannot just allow the endpoint to be always accessed and restrict it to signed-in users only.

**Comment by ajayyy** [comment]
@pgamerx The basic API should be working https://wiki.sponsor.ajay.app/w/API_Docs/Ratings

I haven't added the voting reasons (misleading, educational, funny, etc.) I talked about earlier, but that API change will be backwards compatible.

SponsorBlock doesn't use Oauth, but instead uses a randomly generated userID (https://wiki.sponsor.ajay.app/w/API_Docs#Local_userID_vs_Public_userID).

**Comment by pgamerx** [comment]
> @pgamerx The basic API should be working https://wiki.sponsor.ajay.app/w/API_Docs/Ratings
> 
> I haven't added the voting reasons (misleading, educational, funny, etc.) I talked about earlier, but those changes will be backwards compatible.
> 
> SponsorBlock doesn't use OAuth but instead uses a randomly generated userID (https://wiki.sponsor.ajay.app/w/API_Docs#Local_userID_vs_Public_userID).

That's amazing, one more question - What if someone writes a script that keeps on generating random userIDs and spams the API for a targeted dislike/like (downvote/upvote) attack on a certain video? 

I have another idea, What if we use the Google Oauth string thingy that is returned to us when we use Google API to make someone sign in to anything?
This will increase the work and extra preventive measures would be required to safeguard users because the String can give anyone access to the account.

**Comment by pgamerx** [comment]
Or am I interpreting this wrongly and the user ID can only be generated locally from the "SponsorBlock" extension and not anywhere else?

**Comment by TrashAccount3424** [comment]
@pgamerx If the end-user has to sign-in its rendered useless for a huge portion of user. People dont want a public database that shows what they liked/disliked and a lot also dont want to login to google.

**Comment by ajayyy** [comment]
I don't want to use Google accounts for privacy reasons. 

The current protection is via IP-based ratelimiting. 

There are many ways to still have our own filtering system though. In future, there could be a reputation value calculated with how many reports align with older users to not display votes from new users until they reach that threshold. Would be tricky to figure out the right balance for it, but I highly doubt bots will be a large problem since voting does not affect someone's discoverability, so there is little incentive to abuse it.

**Comment by TrashAccount3424** [comment]
> 
> 
> Or am I interpreting this wrongly and the user ID can only be generated locally from the "SponsorBlock" extension and not anywhere else?

I think different user ID from the same IP are simply discarded

**Comment by pgamerx** [comment]
> @pgamerx If the end-user has to sign in it's rendered useless for a huge portion of users. People don't want a public database that shows what they liked/disliked and a lot also don't want to log in to google.

Then maybe we should add a preventive measure for this, like go generate A Public User ID, you need to use an Authorization key to make sure we know if the script sending the request is a spam script or a legit one. 

The Authorization key should not be provided to everyone but only those who applied through a form and got accepted.

**Comment by pgamerx** [comment]
> > Or am I interpreting this wrongly and the user ID can only be generated locally from the "SponsorBlock" extension and not anywhere else?
> 
> I think different user ID from the same IP are simply discarded

What if they use multiple proxies

**Comment by TrashAccount3424** [comment]
> 
> 
> > @pgamerx If the end-user has to sign in it's rendered useless for a huge portion of users. People don't want a public database that shows what they liked/disliked and a lot also don't want to log in to google.
> 
> Then maybe we should add a preventive measure for this, like go generate A Public User ID, you need to use an Authorization key to make sure we know if the script sending the request is a spam script or a legit one.
> 
> The Authorization key should not be provided to everyone but only those who applied through a form and got accepted.

Would need manual review of thousands of users that's not really practical and a spammer could submit the from automated anyway.

**Comment by pgamerx** [comment]
> > 
> > 
> > > @pgamerx If the end-user has to sign in it's rendered useless for a huge portion of users. People don't want a public database that shows what they liked/disliked and a lot also don't want to log in to google.
> > 
> > Then maybe we should add a preventive measure for this, like go generate A Public User ID, you need to use an Authorization key to make sure we know if the script sending the request is a spam script or a legit one.
> > 
> > The Authorization key should not be provided to everyone but only those who applied through a form and got accepted.
> 
> Would need manual review of thousands of users that's not really practical and a spammer could submit the from automated anyway.

We would only accept their request if they have a project they want to integrate with Sponsorblock like an extension, an application or etc. But this is just my suggestion and I know it's not the best, but I couldn't think of anything else.

**Comment by pgamerx** [comment]
> I don't want to use Google accounts for privacy reasons. 
> 
> The current protection is via IP-based ratelimiting. 
> 
> There are many ways to still have our own filtering system though. In future, there could be a reputation value calculated with how many reports align with older users to not display votes from new users until they reach that threshold. Would be tricky to figure out the right balance for it, but I highly doubt bots will be a large problem since voting does not affect someone's discoverability (unlike social media), so there is little incentive to abuse it.
> 
> SponsorBlock has never had a problem with bot users abusing the API.

Oh I missed your message, this is okay and makes sense. 

The main issue is that there is a high chance that Sponsorblock almost triples the number of the users within few weeks because if youtube doesn't add dislikes back, people will find alternatives and SponsorBlock is the best option by far, because it's used by a lot of people and can already kickstart the rating system.

**Comment by TrashAccount3424** [comment]
> We would only accept their request if they have a project they want to integrate with Sponsorblock like an extension, an application or etc. But this is just my suggestion and I know it's not the best, but I couldn't think of anything else.

A spammer could just use someone elses app then wire-shark the key or de-compile the app and use it to spam.

I dont think any of that is needed. The only people who would have an incentive to spam likes/dislikes(?) would be the creator of a video which who the same time make so bad videos that he need the fake likes. 
Sounds like niche to me.

**Comment by pgamerx** [comment]
Makes sense, kindly ignore whatever I said above.

**Comment by fourteen-1** [comment]
@ajayyy How about taking the YouTube Like-Dislike data from the YouTube API and storing it in the SponsorBlock API? (Just make the dislike number show instead of the word "DISLIKE"). This way a still frame of the Like-Dislike ratio will be preserved (for all videos posted before YT removes the data from their API) and we can determine the value/quality of a video from this archived like-dislike ratio?

**Comment by pgamerx** [comment]
> @ajayyy How about taking the YouTube Like-Dislike data from the YouTube API and storing it in the SponsorBlock API? (Just make the dislike number show instead of the word "DISLIKE"). This way a still frame of the Like-Dislike ratio will be preserved (for all videos posted before YT removes the data from their API) and we can determine the value/quality of a video from this archived like-dislike ratio?

This is what I was doing, but I realised that this isn't like possible because almost 40k videos are posted every hour and this would only work for videos posted before 13th (and that too not all). Maybe @ajayyy can figure out a way to use WayBack machine's API to fetch dislikes?

**Comment by ajayyy** [comment]
Jopik, who runs https://filmot.com/, has a datadump that they are preparing.

**Comment by pgamerx** [comment]
That's amazing, we can utilise that data.

**Comment by fourteen-1** [comment]
@pgamerx 
> This is what I was doing, but I realised that this isn't like possible because almost 40k videos are posted every hour 

What is the primary issue/obstacle in doing this? 

> only work for videos posted before 13th

As someone who uses youtube mainly for learning, the best tutorials on youtube are mostly the older ones. There's innumerable videos of great value already posted and there's still a chance of preserving the best identifier of these great videos in the form of the like/dislike ratio. Without the L/D ratio, finding useful content will be like searching a needle in a haystack.

If there are any financial difficulties in retrieving the like/dislike data from youtube then please by all means set up a crowdfunding system. If you do it right, thousands like me would gladly help to get this very crucial thing done.

**Comment by OpeningCredits** [comment]
What is the use of cataloging the old like to dislike ratio data from the API? There are literally billions of videos on YouTube already, and even if you had the ability to retain all that data, this data would not include “reasoning”, which can only be added after the fact. This seems like potentially a lot of work for something that isn't necessary. We were discussing disliking or liking the video based on a reason, we were not focusing on simply preserving existing data.

**Comment by BlackSails7** [comment]
@OpeningCredits the *reasons*(labels) should include "like" and "dislike" imo, if so, then this way we could* use that data. However it would create "unfair" ratios for the rest of the labels. Therefor they should not be used like that.

On the other hand if they were used for setting youtube's dislike count (replace the 'DISLIKE' text with the number of archived dislikes), the problem with that is that the likes* would continue to increase, whereas the dislikes would stay the same. So it would create inaccurate ratings eventually.

Therefor the only way they could be used, is by showing a tooltip with the archived likes/dislikes somewhere. (perhaps when you hover the like/dislike buttons) And this tooltip could be an optional feature in settings.

**Comment by ajayyy** [comment]
I think multiple ratio bars would make most sense. With hovering showing you raw numbers. Could be configurable too

**Comment by OpeningCredits** [comment]
> I think multiple ratio bars would make most sense. With hovering showing you raw numbers. Could be configurable too

Multiple ratio bars? How would this look?

**Comment by BlackSails7** [comment]
> 
> 
> > I think multiple ratio bars would make most sense. With hovering showing you raw numbers. Could be configurable too
> 
> Multiple ratio bars? How would this look?

now that youtube removed their dislikes, the ratiobar bellow the like/dislike buttons is missing. It can be brought back with the archived data. When this ratiobar is hovered, it should display a tooltip with the archived likes/dislikes and % ratio like it used to.

However the add-on should have the option to hide youtube's likes/dislikes and instead add a like/dislike automatically when you use the addon! This way you can rate both CrowdRate and youtube at the same time. So if the rating you gave was a positive, it would also click youtube's like button. If the rating(label) was negative it would add dislike. Simple.

The question then is... where would the (second) ratiobar go for the archived likes/dislikes? Bellow the first I guess? Which would only appear in old videos? I guess it would be optional, and should also have the option to be thinner (or be thinner by default)

for example (above bar = extension | bellow = archived):

![rating example](https://user-images.githubusercontent.com/44932068/142332826-1d777fdb-7a74-479a-9fa0-6b7732115512.png)

or the archived could replace the thin 1px separator youtube has.. and when hovered it shows more info etc

![rating example 2](https://user-images.githubusercontent.com/44932068/142332994-e35f2a65-e958-417e-b7d0-a0962f8b96ad.png)

**Comment by pgamerx** [comment]
Hey there, I have a question. 
In [This page](https://wiki.sponsor.ajay.app/w/API_Docs/Ratings) of the API docs, what does it mean by "It should be the first 4 - 32 characters (4 is recommended)". Is it telling that the VideoID's only first 4 digits should be converted into SHA256 or only the first 4 characters of the SHA256 string should be provided?

(Sorry I am new to SponsorBlock's API and SHA256 encoding so I have no idea how to work with these, also sorry if this is the wrong place to ask something regarding the API)

**Comment by Jopik1** [comment]
I've made an export of my current data and uploaded to archive.org

https://archive.org/details/Youtube_dislike_count_export_2021-11-18

The file contains YouTube video ids with fetheddate, viewcount, likecount and dislike count.
1,274,565,293 records total out of which 125,395,936 marked as deleted.
Crawled between October 2018 and November 2021.

**Comment by ajayyy** [comment]
> Is it telling that the VideoID's only first 4 digits should be converted into SHA256 or only the first 4 characters of the SHA256 string should be provided?
@pgamerx

The entire videoID needs to be hashed, and 4 characters from the resulting hash should be sent. This wiki page might be helpful https://github.com/ajayyy/SponsorBlock/wiki/K-Anonymity

**Comment by pgamerx** [comment]
Thanks, really helped a lot

**Comment by OpeningCredits** [comment]
> > > I think multiple ratio bars would make most sense. With hovering showing you raw numbers. Could be configurable too
> > 
> > 
> > Multiple ratio bars? How would this look?
> 
> now that youtube removed their dislikes, the ratiobar bellow the like/dislike buttons is missing. It can be brought back with the archived data. When this ratiobar is hovered, it should display a tooltip with the archived likes/dislikes and % ratio like it used to.
> 
> However the add-on should have the option to hide youtube's likes/dislikes and instead add a like/dislike automatically when you use the addon! This way you can rate both CrowdRate and youtube at the same time. So if the rating you gave was a positive, it would also click youtube's like button. If the rating(label) was negative it would add dislike. Simple.
> 
> The question then is... where would the (second) ratiobar go for the archived likes/dislikes? Bellow the first I guess? Which would only appear in old videos? I guess it would be optional, and should also have the option to be thinner (or be thinner by default)
> 
> for example (above bar = extension | bellow = archived):
> 
> ![rating example](https://user-images.githubusercontent.com/44932068/142332826-1d777fdb-7a74-479a-9fa0-6b7732115512.png)
> 
> edit*:
> 
> ![cr-example2](https://user-images.githubusercontent.com/44932068/142345686-bf8b4ba4-624f-4788-8ed0-afe2fd2317b4.png)
> 
> Something similar to this is what I imagine.
> 
> When you click the thumbs up button, it will automatically select the 'like' and popup the options above. Then if you don't change the label within 3 seconds, and your cursor isn't hovering the popup, then it gets hidden automatically. This way if you simply want to 'like' a video, you wont have to click twice!
> 
> Just an idea!

This seems fine. I just hope however it is implemented isn't too resource intensive. Looking at the gradient makes me think that it might use a lot of memory.

**Comment by megapro17** [comment]
@usermyname12 
> ((likes+views/100)/views*1500) i've tested it and it works _ok_ on videos with average views

Uhm, is it really working?
```
>>> likes = 3061749
>>> views = 221502258
>>> ((likes+views/100)/views*1500)
35.733980508677256
```

**Comment by gatosyperros** [comment]
> What about a simple like/dislike system, but when you click like or dislike, it would require a reason by popping up a small panel above the buttons.

This would be a really nice implementation, I personally miss the days where the internet would rate things on a scale from 1 to 5.
Having this option to have more meaning behind a "like"/"dislike" would be really neat.

If you've used YouTube enough you know that they occasionally query the users for why they liked/disliked a video with the exact same question: "I found it: emotional, helpful, entertaining" or if you dislike "boring, annoying, hateful"

For me personally I think just the like dislike ratio would be more of an influence to what videos I'd watch, but I wouldn't mind toying around with the idea of having a fourth option to determine if the video is worth watching.
1. thumbnail, 2. title, 3. like/dislike, 4. a gradient showing in colors what people though.

I don't think you would need any user added titles, descriptions or anything like that, maybe let the users pick from a list of descriptors, and have the average over a bar between the likes and dislikes.

**Comment by Badagronk** [comment]
Personally I'd be happy enough with a simple dislike count! 

The purpose of the dislike system for me was: "There's X views and Y dislikes, despite the subject not being political, so the video is likely full of shit." Especially useful for tutorials and such.

"There's X views and Y sponsorblock-dislikes" would be good enough, if not already better than the retiring system - as someone involved enough to customize their experience likely has a more meaningful dislike than a generic user.

**Comment by pgamerx** [comment]
Hey there, is the API protected/restricted?
Because me and some others were testing the API but kept getting 404 on ` /api/ratings/5f6b`

**Comment by pgamerx** [comment]
Hey, the API doesn't seem to be providing a response.
We tried the following endpoints:
`https://sponsor.ajay.app/api/ratings/AFD2` and `https://sponsor.ajay.app/api/ratings/5f6b`

They both returned `404` along with text: `Cannot GET /api/ratings/AFD2` and `Cannot GET /api/ratings/5f6b` respectively. 

Just to test our theory, we sent some random gibberish as the SHA256 prefix and we didn't get an error, in fact, we got the same error as provided above.
Maybe we are doing something wrong, that's why I am posting it here because I am not sure where we are going wrong.

**Comment by ajayyy** [comment]
@pgamerx https://sponsor.ajay.app/api/ratings/rate/AFD2

**Comment by thereaper90917** [comment]
what does it mean if it response with an empty array? (sorry new to the api)

**Comment by ajayyy** [comment]
It should mean that there hasn't been any submissions with matching videos

**Comment by pgamerx** [comment]
> @pgamerx https://sponsor.ajay.app/api/ratings/rate/AFD2 

Thanks, also there is an issue in the docs then as they say `/api/ratings/sha256` for getting the ratings.

**Comment by OpeningCredits** [comment]
I was thinking about this project for a few days, and it just dawned on me how useful this project could truly become. But I would like to suggest the additional feature of not just being able to dislike a video, but like it as well.

What made me think about this was the fact that I've more frequently been using alternative frontends for youtube these last few weeks. Like Piped and FreeTube. These applications both have Sponsorblock built in, but being privacy-oriented frontends, they lack the ability to like/dislike and comment on videos. This functionality could be added by this new extension we are discussing and add a previously non-existent feature to these alternate frontends.

This could be more of a game changer than I had previously thought.

**Comment by fourteen-1** [comment]
> Personally I'd be happy enough with a simple dislike count!
> 
> The purpose of the dislike system for me was: "There's X views and Y dislikes, despite the subject not being political, so the video is likely full of shit." Especially useful for tutorials and such.
> 
> "There's X views and Y sponsorblock-dislikes" would be good enough, if not already better than the retiring system - as someone involved enough to customize their experience likely has a more meaningful dislike than a generic user.


@Badagronk A simple dislike count is a bad idea. Not everyone who views the video will engage with the like and dislike buttons, in fact, majority of the viewers don't. You'll see videos with a million or two views will at most get 100k-200k likes. So trying to gauge the usefulness/quality of a video by comparing the dislike count to the view count will not work in majority cases. This is especially true for tutorial videos as they get significantly lesser amount of engagement in the like/dislike buttons compared to a music video and other entertainment content. Point is, how much of the viewership contributes to liking/disliking the video is not a constant, so trying to determine the value of a video by comparing dislike-view or like-view ratio is not possible. It has to be the Like-Dislike ratio.

IMO, best thing to do is just freeze the dislike **and like** count as is currently and show that archived ratio. This way we'll have a fair ratio to gauge the value of a video. For future videos something else has to be implemented like Sponsorblock's own like-dislike system. But for older content archiving the L-D count is the best way to do it. 

So this is how it should go:

- Archived Like-Dislike count up until December 2021.
- Real-time Like count of YouTube
- SponsorBlock's own Like-Dislike count.

**Comment by OpeningCredits** [comment]
> So this is how it should go:
> 
>     * Archived Like-Dislike count up until December 2021.
> 
>     * SponsorBlock's own Like-Dislike count.
> 
>     * Real-time Like count of YouTube (Pretty much a useless info without dislike count)

Seems about right to me. @ajayyy ?

**Comment by array-in-a-matrix** [comment]
Why bother archiving the like counter? Isn't dislike enough + real-time like counter?

**Comment by fourteen-1** [comment]
> Why bother archiving the like counter? Isn't dislike enough + real-time like counter?

Giving one factor the ability to increase over time while its counter-agent is frozen in time will create inaccurate and unfair results.

**Comment by Anarios** [comment]
> > Why bother archiving the like counter? Isn't dislike enough + real-time like counter?
> 
> Giving one factor the ability to increase over time while its counter-agent is frozen in time will create inaccurate and unfair results.

Live likes can be displayed (i.e. - don't touch number received by browser from google).

But it's nice to have them saved for analysis.

**Comment by xenago** [comment]
> > > Why bother archiving the like counter? Isn't dislike enough + real-time like counter?
> > 
> > 
> > Giving one factor the ability to increase over time while its counter-agent is frozen in time will create inaccurate and unfair results.
> 
> Live likes can be displayed (i.e. - don't touch the number received by browser from google).
> 
> But it's nice to have them saved for analysis.

I agree. The data is there, it should be presented to users if desired - a toggle for each of the rating options in the settings would solve this and accommodate all preferences.

**Comment by OpeningCredits** [comment]
> Why bother archiving the like counter? Isn't dislike enough + real-time like counter?

This would enable alternate front ends the ability to gain access to a feature they would otherwise not have. For example, the ability to like a youtube video through a community like/dislike feature would be good for privacy as you wouldn't need to log in to YouTube to provide that rating. This like to dislike ratio can supplement the existing one.

**Comment by array-in-a-matrix** [comment]
> > Why bother archiving the like counter? Isn't dislike enough + real-time like counter?
> 
> Giving one factor the ability to increase over time while its counter-agent is frozen in time will create inaccurate and unfair results.

That is very reasonable, nice.

**Comment by ixfd64** [comment]
Another cool feature would be a custom commenting system. It would allow people to discuss videos that have comments disabled. So something like Gab's Dissenter, but hopefully without the alt-right user base.

**Comment by ajayyy** [comment]
I think instead of comments, it should be focused on reviews. There could be text reviews attached to a rating, because I feel it would be more useful than just being a general discussion area.

But, we can see how needed that is over-time, or if a set of rating categories (funny, misleading, etc.) is good enough.

**Comment by pgamerx** [comment]
By the way, out of context but in Youtube Vanced's server, the core dev said that they will be looking forward to adding your (Sponsorblock's) API to Youtube Vanced once it's complete.

**Comment by OpeningCredits** [comment]
> I think instead of comments, it should be focused on reviews. There could be text reviews attached to a rating, because I feel it would be more useful than just being a general discussion area, and we don't have to worry as much about free speech drama.
> 
> But, we can see how needed that is over-time, or if a set of rating categories (funny, misleading, etc.) is good enough. I don't think that should be added for a while to see how it goes.

Sounds good to me.

**Comment by ajayyy** [comment]
![image](https://user-images.githubusercontent.com/12688112/143179252-42434a3c-bd74-4288-b304-13381c07b553.png)


Current list of reasons. Feel free to suggest stuff to add or ways to remove some to condense the list and not overwhelm the user.

**Comment by OpeningCredits** [comment]
> Off Subject = filled with irrelevent content (ie. not concise). Suggestions for other names would be appreciated.

Like, love, funny, informative, interesting, entertaining
Dislike, hate, unfunny, useless, boring, fake/misleading

These would be my suggestions.

**Comment by BlackSails7** [comment]
What if something is useful? For example a tutorial, or information that was life changing? Perhaps replace "accurate" with useful?

I would replace "HighProductionQuality" with Quality ? (because obviously big titles don't fit in the popup options, not enough space)

What about "scam/fraud"? I would replace "meaningless" and "insensitive" (perhaps replace insensitive with ignorant)

I think Misleading Title should be "Misleading Thumbnail / Title" (because so many mainstream videos have bait thumbnails) but I don't know whether putting it together with the misleading title is the right thing to do. It's common to waste your time on a video, because of misleading thumbnail, so I think it's worthy.

Also I'm thinking.. since we have a narcissism epidemic, maybe 'Narcissistic' should be one of the options !? (check out [joeybtoonz](https://www.youtube.com/channel/UCCzK4LJkDl73MQMbr8Lrcww) if you don't believe me) There are too many videos on youtube with severe narcissistic behavior.

Anyway these are my thoughts about the popup options. Dunno.

**Comment by pgamerx** [comment]
I think there can be a category named 'Harmful' or 'dangerous' as well for videos including child abuse, animal abuse, or anything dangerous or harmful.

**Comment by OpeningCredits** [comment]
> I think there can be a category named 'Harmful' or 'dangerous' as well for videos including child abuse, animal abuse, or anything dangerous or harmful.

This is useless. The extension has no communication with YouTube, so it wouldn't aid or help in getting actually harmful material removed. The only way to do that is by reporting it on YouTube itself. This would be a waste of time.

**Comment by BlackSails7** [comment]
> I think there can be a category named 'Harmful' or 'dangerous' as well for videos including child abuse, animal abuse, or anything dangerous or harmful.

maybe this could be a thing, since there are many gray areas youtube doesn't consider child abuse (for example vegan malnutrition) ... there are thousands of youtube channels promoting a strict vegan diet to children and babies, even tho plants are missing 50+ animal nutrients and humans are not herbivorous animals.

And there are some videos showing children with severe disease resulting from severe malnutrition (the parents are blind and in denial), and you can clearly see that the children are starving, but youtube does nothing. [Here](https://old.reddit.com/r/exvegans/comments/k5zfnv/case_reports_of_vegansvegetarians_harming_children/) you can a few reported cases

And this is the same thing for adults and teenagers also.... There are endless vegan channels which are promoting this severe malnutrition to millions, yet youtube allows it, even when it gets EXTREMELY obvious that these channels are promoting severe malnutrition (to teenagers for the most part).

So for this reason alone, I could see use for having a label which would include "child abuse" or "animal abuse" (thousands of vegans feed their dogs/cats a vegan diet nowadays, even tho cats/dogs are also obligate carnivores like humans)

And there are other "minor" child abusive things people think are okay. Others make fun of their pets being obese or deformed.

If a label like this was added, it would have to be something more generic, such as "Abuse" or "Dangerous" so that it includes all kinds of abuse/child abuse/animal abuse.

But I dunno. Is this add-on supposed to bother with this?

**Comment by ajayyy** [comment]
Maybe "disturbing" is more clear? Could replace "insensitive"

**Comment by BlackSails7** [comment]
btw, I'm thinking the add-on should be called "Crowd Rating" instead... because "Rate" also means something else.

Or maybe it should be called "YouTube Reactions" (including "youtube" might make it more popular) dunno

Something else.. the add-on should potentially have the option to store locally any video you ever give rating to (save title/url/thumbnail/ratings, and thus give users a History 'tab' (in options?) for all likes/dislikes you can look at, including deleted videos, and thus allow you to easily modify at any moment (not the deleted ones but the rest)) Shouldn't be an issue with privacy since it would be disabled by default and optional. (also it should be all stored inside a folder, which could be backed up easily, in case someone needed to reinstall (this add-on) somewhere else or whatever)

One reason is because youtube doesn't give a 'dislike' history list, so we have no idea about the total videos we have disliked. And we don't have a like/dislike ratio for ourselves (just for curiosity)

Also there are endless deleted videos in my 'liked playlist' on youtube, and it would be nice to be able to see what they were. That would be often useful. (I have found a way to see what videos the deleted ones were, by copying their URL and searching for it in my Browsing History and thus it gives a title)

Besides, if we don't have accounts when using (this add-on) where will all of our personal ratings be stored? How will we know what type of rating we gave on each video when re-visiting a video? (refreshing tab for example)

**Comment by ixfd64** [comment]
I really like the idea of rating categories!

However, the "outdated" rating should be placed in a special category that does not affect a video's overall rating because videos sometimes become outdated due to reasons beyond the content creator's control.

**Comment by ajayyy** [comment]
```ts
export enum PositiveRatingCategory {
    Like = "like",
    Accurate = "accurate",
    Detailed = "detailed",
    Concise = "concise",
    Insightful = "insightful",
    Impressive = "impressive",
    Funny = "funny",
    Exciting = "exciting"
}

export enum NegativeRatingCategory {
    Dislike = "dislike",
    Incorrect = "incorrect",
    MisleadingTitle = "misleadingTitle",
    OffSubject = "offSubject",
    Outdated = "outdated",
    Disturbing = "disturbing"
}
```

New list

**Comment by ajayyy** [comment]
As suggested on Discord, maybe in the future we could also have neutral tags such as "contains flashing lights", etc. for a crowdsourced warning system. Ideally with predefined categories to help direct the reports to be constructive.

I think the initial version should stick with positive and negative votes though.

**Comment by mchangrh** [comment]
possibly as content flags - disturbing might also fit in there?

content with ...
- photosensitivity warning
- disturbing content

**Comment by ajayyy** [comment]
Could also allow users to configure which labels they consider negative

**Comment by DinuruSeniya** [comment]
Glad to see work is being done in this area. YouTube is slowly being monopolized. It's upto the community now to fix it.

**Comment by OpeningCredits** [comment]
> ```ts
> export enum PositiveRatingCategory {
>     Like = "like",
>     Accurate = "accurate",
>     Detailed = "detailed",
>     Concise = "concise",
>     Insightful = "insightful",
>     Impressive = "impressive",
>     Funny = "funny",
>     Exciting = "exciting"
> }
> 
> export enum NegativeRatingCategory {
>     Dislike = "dislike",
>     Incorrect = "incorrect",
>     MisleadingTitle = "misleadingTitle",
>     OffSubject = "offSubject",
>     Outdated = "outdated",
>     Disturbing = "disturbing"
> }
> ```
> 
> New list

Accurate, detailed and concise as different options seems a bit overkill to me. I feel like having this many options isn't necessarily a bad thing, but it depends on how it shows up. If these options show up as a kind of tooltip that shows up when you like/dislike a video, then I think it's okay. But it would be best not to have all these options permanently on screen.

**Comment by ajayyy** [comment]
I've been thinking if adding Disagree and Agree as well to separate them from the more objective answers and allow you to ignore opinion answers if they don't matter to you

**Comment by OpeningCredits** [comment]
> I've been thinking if adding Disagree and Agree as well to separate them from the more objective answers and allow you to ignore opinion answers if they don't matter to you

That makes sense.

**Comment by fourteen-1** [comment]
> I've been thinking if adding Disagree and Agree as well to separate them from the more objective answers and allow you to ignore opinion answers if they don't matter to you

Yes, most people want to determine the worth of a video from glancing at two ratings. The L/D ratio did brilliant for many years for YT. Analyzing 5-6 different categories two times over might be off-putting or overwhelming as you said for the average user. Its a good feature for power users. 

Having the sum of all categories of one rating show as the "total" for that rating will be neat. Total (+)/Like/Agree rating and (-)/Dislike/Disagree Rating should be positioned somewhere at the top of the detailed category ratings.

**Comment by ajayyy** [comment]
Yep, it could also be useful to show the text of the category with the most votes.

**Comment by lockHrt** [comment]
Came here to suggest this very feature.
Here's an idea from my side: we tap into dislike button so when a user dislikes a video, the dislike request is sent to YouTube server as well sponsor block server.

We will then have following:
1. total views (V)
2. total views by sponsor block users. (sV)
3. number of likes (L)
4. number of dislikes by sponsor block users (sD)

So calculating it would look something like this,
sD / [(sV / V * L) + sD] = Like/Dislike ratio

For example, let's say we have these values,
1. total views (V) =  10,000
2. total views by sponsor block users. (sV) = 1,000
3. number of likes (L) = 500
4. number of dislikes by sponsor block users (sD) = 30

30 / [(1,000 / 10,000 * 500) + 30] = **62%**

In the end, instead of previously showing users like/dislike in counts, it'll show in percentage. It'll be very close.

**Comment by BlackSails7** [comment]
I think we should not focus so much on the options right now and stop changing them every single week, because they can change anytime in the future anyway. All you would need to do in order to change a label is to transfer all the values from the options(labels you want to remove) into the likes or dislikes.

This way we can easily change a label to something completely different, without creating inaccurate ratings on videos as the result.

I think we will see over time and testing (within the first month after release) which labels need to be modified/replaced, depending on how many users apply them and whether they are misused. There is no way the labels wont change over time.

We also need a description for each option(label) to explain in *detail* what it really means. (we can also show a tooltip for each label when we open the ratiobar-tooltip and _hover_ the label-texts in it (you can lock the ratiobar-tooltip by clicking it)) This way we would prevent some percentage of misuse.

For example the "disturbing" label inside the ratiobar-tooltip.. when you hover it, it would show a tooltip and show every category it includes, this way it would become less subjective, because users would be discouraged from using it for things that aren't included. If they think that there are more things it should include, then they are free to request as long as they provide valid arguments.

But needless to say, it will never be perfect. People will always give illogical ratings based on their feelings and cognitive dissonance, there is no way around that.

**Comment by ajayyy** [comment]
@lockHrt https://returnyoutubedislike.com/ will already be doing something like that, I think this one should focus on meaningful categorisation and reporting.

**Comment by NC22** [comment]
@ajayyy Hi. Also added implementation of SponsorBlock Ratings API from https://wiki.sponsor.ajay.app/w/API_Docs/Ratings to my project https://github.com/NC22/KellyC-Return-YouTube-Dislikes for testing purposes, disabled it for prod versions. (need to uncomment last string in extension/lib/api/sponsorBlock.js for load to providers pool)

may be that would be helpfull for some one who test it also. 

Currently tested it as separate data source, seem works ok. One time got response from GET /api/ratings/rate/ with count = -1 for type = 0 in response. That possibly happen because over spam of POST /api/ratings/rate with enabled = false during tests without actual checks. not tried to repeat that later.

For now dont have any plans on that, but I think its good to have several data sources for compare with or even combine them, if they be more alife in future. If api is ready atleast in basic way, I can optionaly allow users to send actions to this api already.

Idea of reaction \ reporting popups on like \ dislike actions also looks interesting - https://github.com/ajayyy/SponsorBlock/issues/1039#issuecomment-970572394 will look on next api updates

**Comment by ajayyy** [comment]
@NC22 I changed it to write to the private db first, maybe it was a race condition and it allowed two undo votes at once.

https://github.com/ajayyy/SponsorBlockServer/commit/a0720329d0633153ffa0fc98c041127eade3fbe1

**Comment by BlackSails7** [comment]
Crowdrate-preview v2 is finally finished!

Now has light theme, colorpicker, and 4x more options

**Download:** https://www.mediafire.com/file/on4xsn5x6tmljbh/WindowsNoEditor_v2.7z/file

![Screenshot (27)](https://user-images.githubusercontent.com/44932068/145918844-cd7ea495-83fe-4b97-b0de-aab7328773c6.png)
![Screenshot (28)](https://user-images.githubusercontent.com/44932068/145919312-eef55157-54de-470a-8183-9df01a850771.png)
![Screenshot (29)](https://user-images.githubusercontent.com/44932068/145919324-06782ba6-dd5d-453d-858e-0b2a6a0e6690.png)

**Comment by GlowingUmbreon** [comment]
> Crowdrate-preview v2 is finally finished!
> 
> Now has light theme, colorpicker, and 4x more options
> 
> **Download:** https://www.mediafire.com/file/pmxkflgnqrbsm3c/CrowdRate_v2.7z/file (43MB)
> 
> ![Screenshot (27)](https://user-images.githubusercontent.com/44932068/145918844-cd7ea495-83fe-4b97-b0de-aab7328773c6.png) ![Screenshot (28)](https://user-images.githubusercontent.com/44932068/145919312-eef55157-54de-470a-8183-9df01a850771.png) ![Screenshot (29)](https://user-images.githubusercontent.com/44932068/145919324-06782ba6-dd5d-453d-858e-0b2a6a0e6690.png)

This is malware, DO NOT USE!

**Comment by GlowingUmbreon** [comment]
> Crowdrate-preview v2 is finally finished!
> 
> Now has light theme, colorpicker, and 4x more options
> 
> **Download:** https://www.mediafire.com/file/pmxkflgnqrbsm3c/CrowdRate_v2.7z/file (43MB)

You are distributing a youtube plugin as a totally not dodgy looking .exe file made by unreal engine?

**Comment by ajayyy** [comment]
It's not a plugin, it's a demo of their design

**Comment by GlowingUmbreon** [comment]
but as a Unreal engine executable? This is one of the most suspicious things I've seen in a long while.

**Comment by BlackSails7** [comment]
> But as a Unreal engine executable? This is one of the most suspicious things I've seen in a long while. If its a preview just post some screenshots, publish as a psd or maybe even a html mockup, but not as a executable.

hahahahahahah oh no! you caught me XDDD

Do you realize that UE4 has the best UI editor in the world (plus the graph editor for visual scripting making it 100x easier/faster)?. 

I made this in 2 weeks passively, whereas to make it for an add-on will take 3 months or so. And I already released a previous version weeks ago. I would have finished this earlier, but I was busy with other things.

I thought it would take me 1-2 days to make this preview (but as always it was 10x more complicated than I imagined) and we came up with many more ideas/features and settings for it. (which I knew would happen, hence this is the reason I made it)

**Comment by ajayyy** [comment]
I've updated the title of this issue to clarify the purpose of this project.

https://returnyoutubedislike.com/ is the addon for trying to emulate the old dislike functionality

**Comment by ccuser44** [comment]
How would you stop bots abd abuse with this?

**Comment by NC22** [comment]
@ccuser44 as I see u cant post more that one action per video from one IP even if you change user Id hash (u need to cancel prev state, then send new)

https://github.com/ajayyy/SponsorBlockServer/blob/a0720329d0633153ffa0fc98c041127eade3fbe1/src/routes/ratings/postRating.ts#L41

it would be interesting to know, may be there is some addition limitations

**Comment by ccuser44** [comment]
> @ccuser44 as I see u cant post more that one action per video from one IP even if you change user Id hash (u need to cancel prev state, then send new)
> 
> https://github.com/ajayyy/SponsorBlockServer/blob/a0720329d0633153ffa0fc98c041127eade3fbe1/src/routes/ratings/postRating.ts#L41
> 
> it would be interesting to know, may be there is some addition limitations

You could quite simply use proxies, you can get like 3000 proxies for 1$

**Comment by ccuser44** [comment]
Also there are specific extensions for this, I don't think sponsorblock should do this as there are extensions thatvalready do this very good

**Comment by pgamerx** [comment]
Any updates on the development?

**Comment by NC22** [comment]
@ccuser44 What? currently there is no any extension that can protect from spam videos like that https://catface.ru/userfiles/media/udata_1640279852_eoowkelc.png likes counter always be bigger that dislikes, because youtube not protected from this type of spam.

**Comment by zopieux** [comment]
>youtube itself not protected from this type of spam

YouTube likes and dislikes are gated by having a valid Google account. Because Google has massive resources, they can afford entire teams of anti-abuse engineers and curators to mass-ban fake accounts. While imperfect, the net result is that Google/YT accounts are [among the most expensive to buy from fake account websites](https://www.bulkaccountsbuy.com/shop/page/2/).

Creating a crowdsourced service is easy. Creating a *spam-free* crowdsourced service requires lots of resources, or building on top of existing well-funded giants, eg. by adding a captcha or authentication against a mostly-curated identity provider.

**Comment by anon238** [comment]
> I've updated the title of this issue to clarify the purpose of this project.
> 
> https://returnyoutubedislike.com/ is the addon for trying to emulate the old dislike functionality

Is this also coming to SponsorBlock for iOS? I have seen another app that’s using the dislike functionality is he using SponsorBlock as API?

**Comment by ajayyy** [comment]
I'm going to close this issue, as I think "Return YouTube Dislike" covers a lot of what people want, and they have expressed interest in expanding their rating system past just like and dislike, so some of this stuff might be able to be accomplished there.

I'm not ruling out revisiting the CrowdRate system in the future though
